File: leetcode/p16.py
class Solution:
    def threeSumClosest(self, nums, target):
        min = None
        diffMin = None
        for i in range(len(nums) - 2):
            for j in range(i + 1, len(nums) - 1):
                for k in range(j + 1, len(nums)):
                    sum = nums[i] + nums[j] + nums[k]
                    if diffMin == None or diffMin > abs(target - sum):
                        diffMin = abs(target - sum)
                        min = sum
        return min
print(Solution().threeSumClosest([-1, 2, 1, -4], 1))

File: leetcode/p16.js
var threeSumClosest = function(nums, target) {
    let min = null
    let diffMin = null
    for (let i = 0; i < nums.length - 2; i++) {
        for (let j = i + 1; j < nums.length - 1; j++) {
            for (let k = j + 1; k < nums.length; k++) {
                const sum = nums[i] + nums[j] + nums[k]
                if (diffMin === null || diffMin > Math.abs(target - sum)) {
                    diffMin = Math.abs(target - sum)
                    min = sum
                }
            }
        }
    }
    return min
}

console.log(threeSumClosest([-1, 2, 1, -4], 1))